packages/core/src/odata/v2/entity-deserializer.ts

```diff
@@ -4,19 +4,19 @@ import {
   entityDeserializer,
   EntityDeserializer
 } from '../common/entity-deserializer';
-import { edmToTs } from './payload-value-converter';
-import { extractODataETag } from './extract-odata-etag';
+import { edmToTsV2 } from './payload-value-converter';
+import { extractODataEtagV2 } from './extract-odata-etag-v2';
 import { extractDataFromOneToManyLink } from './extract-data-from-one-to-many-link';
 
 /**
  * Entity deserializer instance for v2 entities.
  * See [[EntityDeserializerType]] for the provided methods.
  */
 const deserializer: EntityDeserializer = entityDeserializer(
-  edmToTs,
-  extractODataETag,
+  edmToTsV2,
+  extractODataEtagV2,
   extractDataFromOneToManyLink
 );
 
-export const deserializeEntity = deserializer.deserializeEntity;
-export const deserializeComplexType = deserializer.deserializeComplexType;
+export const deserializeEntityV2 = deserializer.deserializeEntity;
+export const deserializeComplexTypeV2 = deserializer.deserializeComplexType;
```

Comment:
breaking changes

Reply:
after discussing also with marika, we add the export `deserializeEntity` to ensure compatibility since only the ` entityDeserializer` was @experimental.

Reply:
adjusted